packaging/utils.py

```diff
@@ -24,6 +24,10 @@ def canonicalize_version(version):
 
     version = parse(version)
 
+    # Legacy versions cannot be normalized
+    if isinstance(version, LegacyVersion):
```

Comment:
I wonder if this would be better written as:

```python
try:
    version = Version(version)
except InvalidVersion:
    return version
```

With the idea that the ``parse()`` function could hypothetically return more version types in the future if we ever have a PEP440v2 kind of thing. Either way I think it's fine though, so if you prefer it this way I'm happy to merge it.

Reply:
Makes sense to me. I've updated the PR.